pkg/api/controller.go

```diff
@@ -2441,14 +2441,18 @@ func (c *Controller) ListObjects(w http.ResponseWriter, r *http.Request, reposit
 			if !entry.CreationDate.IsZero() {
 				mtime = entry.CreationDate.Unix()
 			}
-			objList = append(objList, ObjectStats{
+			objStat := ObjectStats{
 				Checksum:        entry.Checksum,
 				Mtime:           mtime,
 				Path:            entry.Path,
 				PhysicalAddress: qk.Format(),
 				PathType:        entryTypeObject,
 				SizeBytes:       Int64Ptr(entry.Size),
-			})
+			}
+			if (params.UserMetadata == nil || *params.UserMetadata) && entry.Metadata != nil {
```

Comment:
make sure that for 'ls' if the default is passed by the generated code - as far as I remember the oapi middleware just verify that the request and response are valid.
The default behaviour when there is no value is handled by the code - which means that if 'params.UserMetadata' is nil it is also true.

Reply:
OAPI doesn't handle the default values, you're right, so setting the defaults here doesn't actually do anything, which I find strange. 

While I was testing the call, if the `user_metadata` query param wasn't set in the request, this function would dereference the nil pointer, and the code would throw a runtime error, so I believe this check is needed. With this check, the default behavior I described (returning the user-defined metadata) occurs. 

Since OAPI defaults don't actually set a default, but the behavior is coded here instead, would it be okay to leave the defaults in the Swagger file to act as documentation for these api calls?

Reply:
Got it. Leave the default in the swagger as it can affect the client-side code generated.